fireapp/lib/presentation/calendar/calendar_widget.dart

```diff
@@ -44,20 +35,346 @@ class _CalendarState extends FireAppState<CalendarView>
     implements ViewModelHolder<CalendarViewModel> {
   @override
   CalendarViewModel viewModel = GetIt.instance.get();
+  final List<Color> colorPalette = [
+    const Color(0xFFB9DEEC), // Light Blue
+    const Color(0xFFCBCBEA), // Lavender
+    const Color(0xFFFDD2D7), // Pink
+    const Color(0xFFFCFCDC), // Light Yellow
+    const Color(0xFFEFEFEF), // Light Gray
+    const Color(0xFFF6CEBC), // Light Orange
+    const Color(0xFFCFDEE3),
+    const Color(0xFFCBE4F1),
+  ];
+  int _nextColorIndex = 0;
+
+  final Map<String, Color> eventColorMap = {};
+  // Map a colour for the corresponding event
+  Color? getColorForEvent(String eventId) {
+    if (!eventColorMap.containsKey(eventId)) {
+      eventColorMap[eventId] = colorPalette[_nextColorIndex];
+      _nextColorIndex = (_nextColorIndex + 1) % colorPalette.length;
+    }
+    return eventColorMap[eventId];
+  }
+
+  late DateTime _selectedDate;
+  late String _selectedMonthLabel;
 
   @override
-  void handleNavigationEvent(CalendarNavigation event) {
+  void initState() {
+    super.initState();
+    // Load events and selected month
+    viewModel.loadAndSetDisplayEvents();
+    _selectedDate =
+        DateTime(viewModel.selectedYear.value, viewModel.selectedMonth.value);
+    _selectedMonthLabel = DateFormat('MMM yyyy').format(_selectedDate);
+  }
+
+  @override
+  void handleNavigationEvent(CalendarNavigation navEvent) {
+    // //Navigate to the form, and reload when return
+    navEvent.when(eventDetail: (event) {
+      Navigator.of(context)
+          .push(
+        MaterialPageRoute(
+          builder: (context) => UnavailabilityFormPage(event: event),
+        ),
+      )
+          .then((result) {
+        if (result == 'reload') {
+          viewModel.loadAndSetDisplayEvents();
+        }
+      });
+    });
+  }
+
+// Group all filtered display events to their day
+  Map<DateTime, List<CalendarEvent>> groupEventsByDate(
+      List<CalendarEvent> events) {
```

Comment:
When the parameters of a function are too long to fit on a new line, they should be split and defined one-per-line, with any paratheses on a new line as well. E.g:
```dart
Map<DateTime, List<CalendarEvent>> groupEventsByDate(
  List<CalendarEvent> events,
  String otherHypotheticalParameter
) {
```